pkg/module/sync/runner_impl.go

```diff
@@ -66,134 +64,119 @@ func (r *RunnerImpl) Sync(ctx context.Context, kyma *v1alpha1.Kyma,
 			return fmt.Errorf("error setting owner reference on component CR of type: %s for resource %s %w",
-				name, kyma.Name, err)
+				module.GetName(), kyma.Name, err)
 		}
 	}
 
 	return nil
 }
 
 func (r *RunnerImpl) SyncModuleStatus(ctx context.Context, kyma *v1alpha1.Kyma, modules common.Modules) bool {
-	statusMap := kyma.GetModuleStatusMap()
-	statusUpdateRequiredFromUpdate := r.updateModuleStatusFromExistingModules(modules, statusMap, kyma)
-	statusUpdateRequiredFromDelete := r.deleteNoLongerExistingModuleStatus(ctx, statusMap, kyma)
+	statusUpdateRequiredFromUpdate := r.updateModuleStatusFromExistingModules(modules, kyma)
+	statusUpdateRequiredFromDelete := r.deleteNoLongerExistingModuleStatus(ctx, kyma)
 	return statusUpdateRequiredFromUpdate || statusUpdateRequiredFromDelete
 }
 
-func (r *RunnerImpl) updateModuleStatusFromExistingModules(modules common.Modules,
-	moduleStatusMap map[string]*v1alpha1.ModuleStatus, kyma *v1alpha1.Kyma,
-) bool {
+func (r *RunnerImpl) updateModuleStatusFromExistingModules(modules common.Modules, kyma *v1alpha1.Kyma) bool {
 	updateRequired := false
-	for name := range modules {
-		module := modules[name]
-		descriptor, _ := module.Template.Spec.GetUnsafeDescriptor()
+	for idx := range modules {
+		module := modules[idx]
+		manifestAPIVersion, manifestKind := module.Object.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
+		templateAPIVersion, templateKind := module.Template.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
 		latestModuleStatus := v1alpha1.ModuleStatus{
-			ModuleName: module.Name,
-			Name:       module.Manifest.GetName(),
-			Namespace:  module.Manifest.GetNamespace(),
-			Generation: module.Manifest.GetGeneration(),
-			TemplateInfo: v1alpha1.TemplateInfo{
-				Name:       module.Template.Name,
-				Namespace:  module.Template.Namespace,
-				Channel:    module.Template.Spec.Channel,
-				Generation: module.Template.Generation,
-				GroupVersionKind: metav1.GroupVersionKind{
-					Group:   manifestV1alpha1.GroupVersionKind.Group,
-					Version: manifestV1alpha1.GroupVersionKind.Version,
-					Kind:    manifestV1alpha1.GroupVersionKind.Kind,
-				},
-				Version: descriptor.Version,
+			Name:    module.ModuleName,
+			FQDN:    module.FQDN,
+			State:   stateFromManifest(module.Object),
+			Channel: module.Template.Spec.Channel,
+			Version: module.Version,
+			Manifest: v1alpha1.TrackingObject{
+				PartialMeta: v1alpha1.PartialMetaFromObject(module.Object),
+				TypeMeta:    metav1.TypeMeta{Kind: manifestKind, APIVersion: manifestAPIVersion},
+			},
+			Template: v1alpha1.TrackingObject{
+				PartialMeta: v1alpha1.PartialMetaFromObject(module.Template),
+				TypeMeta:    metav1.TypeMeta{Kind: templateKind, APIVersion: templateAPIVersion},
 			},
-			State: stateFromManifest(module.Manifest),
 		}
-		moduleStatus, exists := moduleStatusMap[module.Name]
-		if exists {
-			if moduleStatus.State != latestModuleStatus.State {
+		if len(kyma.Status.Modules) < idx+1 {
```

Comment:
tbh, I haven't get the point of why needs to distinguish these two cases, but since we don't need to detect if update required, I think we can simplify to only append latestModuleStatus to `kyma.Status.Modules`

Reply:
if you do an append here instead of indexed access, we will not override the existing status. Thus we need it to make sure we do not simply add more elements to the slice every time